appengine/channel/chatactoe.py

```diff
@@ -0,0 +1,210 @@
+
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Channel Tic Tac Toe
+This module demonstrates the App Engine Channel API by implementing a
+simple tic-tac-toe game.
+For more information, see the README.md.
+"""
+
+# [START all]
+
+import datetime
+import logging
+import os
+import random
+import re
+import json
+import jinja2
+import webapp2
+from google.appengine.api import channel
+from google.appengine.api import users
+from google.appengine.ext import db
+
+
+CLOUD_PROJECT_ID = 'your_project_id'
```

Comment:
This can be ascertained with the app engine identity API: `google.appengine.api.app_indentity.get_application_id()`.